### ISSUE TITLE
I tried to convert my dataset into federated dataset. What is the best way to do so? And, how do I know if the data is federated or not. I used a couple of procedures on stack overflow and obtained the result, however I can't differentiate if the data has been changed to federated or not.

### ISSUE THREAD

**Issue by bibek948** [tensorflow/federated#858]
**Describe the bug**
A clear and concise description of what the bug is. It is often helpful to
provide a link to a [colab](https://colab.research.google.com/) notebook that
reproduces the bug.

**Environment (please complete the following information):**
* OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
* Python package versions (e.g., TensorFlow Federated, TensorFlow):
* Python version:
* Bazel version (if building from source):
* CUDA/cuDNN version:
* What TensorFlow Federated execution stack are you using?

Note: You can collect the Python package information by running `pip3 freeze`
from the command line and most of the other information can be collected using
TensorFlows environment capture
[script](https://github.com/tensorflow/tensorflow/blob/master/tools/tf_env_collect.sh).

**Expected behavior**

**Additional context**

**Comment by roselander** [comment]
Hi, can you please fill in the information from the bug template? In particular, it would be helpful to see your code so far.

**Comment by bibek948** [comment]
#the code are as follows


!pip install --quiet --upgrade tensorflow_federated
import collections
import tensorflow as tf
import tensorflow_federated as tff
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split 
from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.preprocessing import LabelEncoder

%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

from google.colab import files
uploaded=files.upload()

import io

#also I rename the kaggle autism dataset to  'autism' 
# the link to the dataset - https://www.kaggle.com/fabdelja/autism-screening-for-toddlers

autismData=pd.read_csv(io.BytesIO(uploaded['autism.csv']))

autismData['Asum']=autismData.A1+autismData.A2+autismData.A3+autismData.A4+autismData.A5+autismData.A6+autismData.A7+autismData.A8+autismData.A9+autismData.A10;
mapping={autismData.columns[18]:'ASD'}

autismData=autismData.rename(columns=mapping)

# here I am dropping unnecessary categorical variable ,
newData=autismData.drop(['A1','A2','A3','A4','A5','A6','A7','A8','A9','A10','Age_Mons','Qchat-10-Score','Sex','Ethnicity','Jaundice','Family_mem_with_ASD','Who completed the test'],axis=1)

#here I am changing my  remaining column data to dummies
autismData = pd.get_dummies(newData, prefix_sep='_', drop_first=True)


#the following code is for changing into federated dataset
client_id_columns='Case_No';
SHUFFLE_BUFFER = 1000
NUM_EPOCHS = 1

#Split client id into train and test clients

client_ids=autismData[client_id_columns]
train_client_ids=client_ids.sample(frac=0.5).tolist()
test_client_ids = [x for x in client_ids if x not in train_client_ids]

#this function accepts client_id as input and return tf.data.Dataset
def create_tf_dataset_for_client_fn(client_id):
  # a function which takes a client_id and returns a
  # tf.data.Dataset for that client
  client_data = autismData[autismData[client_id_columns] == client_id]
  dataset = tf.data.Dataset.from_tensor_slices(client_data.to_dict('list'))
  dataset = dataset.shuffle(SHUFFLE_BUFFER).batch(2).repeat(NUM_EPOCHS)
  return dataset

#now we use the function above to create object for our training and test data
train_data=tff.simulation.ClientData.from_clients_and_fn(
    client_ids=train_client_ids, create_tf_dataset_for_client_fn=create_tf_dataset_for_client_fn
)
test_data=tff.simulation.ClientData.from_clients_and_fn(
    client_ids=test_client_ids,create_tf_dataset_for_client_fn=create_tf_dataset_for_client_fn
)

#now printing one of the data set
example_dataset = train_data.create_tf_dataset_for_client(
    train_data.client_ids[0])

print(example_dataset)
example_element = next(iter(example_dataset))
example_element['Asum'].numpy()


#the output of the above code is
" <RepeatDataset shapes: {Case_No: (None,), Asum: (None,), ASD_Yes: (None,)}, types: {Case_No: tf.int32, Asum: tf.int32, ASD_Yes: tf.int32}>
array([5], dtype=int32)"

But I am confused now. Is this data even federated? If it is, how?  I read in a paper that federated data should be non iid but how do I know looking at this data that it's federated.

**Comment by harshita1804** [comment]
did you find a solution to this? Even I am kind of stuck with a similar problem

**Comment by harshita1804** [comment]
@bibek948 
https://github.com/CS-savvy/experiment_with_tf_federated 

I found this repo really helpful, maybe you could take a look and see if it helps

**Comment by bibek948** [comment]
@harshita1804 thank you for sharing the repo. I paused my project because I was not getting anywhere. Hopefully, now I can continue.

**Comment by nidhitiwari11998** [comment]
@bibek948 did you find how to convert a dataset into federated dataset ? If so, could you please share the method or useful links ?
